ipatests/test_integration/test_idp.py

```diff
@@ -97,20 +102,199 @@ def install(cls, mh):
+        self.master.run_command(["ipa", "hbacrule-add", "rule1"])
+        self.master.run_command(["ipa", "hbacrule-add-user", "rule1",
+                                 "--users=keycloakuser"]
+                                )
+        self.master.run_command(["ipa", "hbacrule-add-host", "rule1",
+                                 "--hosts", self.replica.hostname])
+        self.master.run_command(["ipa", "hbacrule-add-service", "rule1",
+                                 "--hbacsvcs=sshd"]
+                                )
+        tasks.clear_sssd_cache(self.master)
+        tasks.clear_sssd_cache(self.replica)
+        yield
+
+        # cleanup
+        tasks.kinit_admin(self.master)
+        self.master.run_command(["ipa", "hbacrule-enable", "allow_all"])
+        self.master.run_command(["ipa", "hbacrule-del", "rule1"])
+
+    def test_auth_hbac(self, hbac_setup_teardown):
+        """
+        Test case to check that hbacrule is working as
+        expected for user configured with external idp.
+        """
+        kinit_idp(self.master, 'keycloakuser', keycloak_server=self.client)
+        ssh_cmd = "ssh -q -K -l keycloakuser {0} whoami"
+        valid_ssh = self.master.run_command(
+            ssh_cmd.format(self.replica.hostname))
+        assert "keycloakuser" in valid_ssh.stdout_text
+        negative_ssh = self.master.run_command(
+            ssh_cmd.format(self.master.hostname), raiseonerr=False
+        )
+        assert negative_ssh.returncode == 255
+
+    def test_auth_sudo_idp(self):
+        """
+        Test case to check that sudorule is working as
+        expected for user configured with external idp.
+        """
+        tasks.kdestroy_all(self.master)
+        tasks.kinit_admin(self.master)
+        #  rule: keycloakuser are allowed to execute yum on
+        #  the client machine as root.
+        cmdlist = [
+            ["ipa", "sudocmd-add", "/usr/bin/yum"],
+            ["ipa", "sudorule-add", "sudorule"],
+            ['ipa', 'sudorule-add-user', '--users=keycloakuser',
+             'sudorule'],
+            ['ipa', 'sudorule-add-host', '--hosts',
+             self.client.hostname, 'sudorule'],
+            ['ipa', 'sudorule-add-runasuser',
+             '--users=root', 'sudorule'],
+            ['ipa', 'sudorule-add-allow-command',
+             '--sudocmds=/usr/bin/yum', 'sudorule'],
+            ['ipa', 'sudorule-show', 'sudorule', '--all']
+        ]
+        for cmd in cmdlist:
+            self.master.run_command(cmd)
+        tasks.clear_sssd_cache(self.master)
+        tasks.clear_sssd_cache(self.client)
+        try:
+            cmd = 'sudo -ll -U keycloakuser'
+            test = self.client.run_command(cmd).stdout_text
+            assert "User keycloakuser may run the following commands" in test
+            assert "/usr/bin/yum" in test
+            list_fail = self.master.run_command(cmd).stdout_text
+            assert "User keycloakuser is not allowed to run sudo" in list_fail
```

Comment:
I would also try to run the `sudo yum` command as the user `keycloakuser` after authenticating with the external IdP.